src/components/Input/Input.jsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { v4 as uuid } from 'uuid';
+import './Input.scss';
+
+const Input = (props) => {
+  const {
+    className,
+    id,
+    isDisabled,
+    label,
+    onBlur,
+    onChange,
+    onFocus,
+    placeholder,
+    type,
+    value,
+  } = props;
+
+  const handleChange = (event) => {
+    onChange(event);
+  }
+
+  const handleFocus = (event) => {
+    if (onFocus) onFocus(event);
+  }
+
+  const handleBlur = (event) => {
+    if (onBlur) onBlur(event);
+  }
+
+  const classes = classNames(
+    className,
+    'input',
+    // { @TODO Placeholder until size/color props are added.
+    //   [`font-size-${size}`]: size,
+    //   [`font-color-${color}`]: color,
+    // }
+  );
+
+  const inputId = id || uuid();
+
+  const renderInput = () => (
+    <input
+      className={classes}
+      disabled={isDisabled}
+      id={inputId}
+      onBlur={handleBlur}
+      onChange={handleChange}
+      onFocus={handleFocus}
+      placeholder={placeholder}
+      type={type}
+      value={value}
+    />
+  );
+
+  const renderLabel = () => {
+    if (!label) return;
+
+    return (
+      <label className="label" htmlFor={inputId}>
+        {label}<span className="font-color-danger font-size-sm">&nbsp;*</span>
+      </label>
+    );
+  }
+
+  return (
+    label ? (
+      <>
```

Comment:
thats a cool trick, didnt know about this (is it just a div)?

Reply:
@ericyliu this one is a fragment actually, it will just disappear on render so only your elements go in the dom.